src/components/ItemSummary.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { ItemData } from "../LibraryUtilities";
+import { LibraryContainer } from "./LibraryContainer";
+
+interface ItemSummaryProps {
+    libraryContainer: LibraryContainer,
+    data: ItemData;
+    showDescription: boolean;
+}
+
+interface ItemSummaryStates {
+    hasSummaryData: boolean;
+}
+
+export class ItemSummary extends React.Component<ItemSummaryProps, ItemSummaryStates> {
+    summaryData: any;
+
+    constructor(props: ItemSummaryProps) {
+        super(props);
+        this.state = ({ hasSummaryData: false });
+        this.summaryData = null;
+        this.onDataReceived = this.onDataReceived.bind(this);
+    }
+
+    render() {
+        this.onLibraryItemSummaryExpand();
```

Comment:
`onLibraryItemSummaryExpand` should be renamed to `fetchMissingItemSummary` so that it does not sound like it will raise an event for external components to do other things.